spring-cloud-data-admin/src/main/java/org/springframework/cloud/data/admin/controller/ModuleController.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.admin.controller;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.data.core.ModuleCoordinates;
+import org.springframework.cloud.data.core.ModuleType;
+import org.springframework.cloud.data.module.registry.ModuleRegistration;
+import org.springframework.cloud.data.module.registry.ModuleRegistry;
+import org.springframework.cloud.data.rest.resource.DetailedModuleRegistrationResource;
+import org.springframework.cloud.data.rest.resource.ModuleRegistrationResource;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.web.PagedResourcesAssembler;
+import org.springframework.hateoas.ExposesResourceFor;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Handles all Module related interactions.
+ *
+ * @author Glenn Renfro
+ * @author Mark Fisher
+ * @author Gunnar Hillert
+ * @author Eric Bottard
+ * @author Gary Russell
+ * @author Patrick Peralta
+ */
+@RestController
+@RequestMapping("/modules")
+@ExposesResourceFor(ModuleRegistrationResource.class)
+public class ModuleController {
+
+	private final Assembler moduleAssembler = new Assembler();
+
+	private final ModuleRegistry registry;
+
+	@Autowired
+	public ModuleController(ModuleRegistry registry) {
+		this.registry = registry;
+	}
+
+	/**
+	 * List module registrations.
+	 */
+	@RequestMapping(value = "", method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	@ResponseBody
```

Comment:
`@ResponseBody` is redundant when using `@RestController` for the class. also, we can remove `value = ""` in the RequestMapping (ok to be in there for clarity).